runtime/onert/backend/train/KernelGenerator.cc

```diff
@@ -85,8 +94,9 @@ std::unique_ptr<exec::train::TrainableFnSequence> KernelGenerator::generate(ir::
   assert(_return_fn);
   ret->append(std::move(_return_fn));
 
-  if (_update_fn)
-    ret->append(std::move(_update_fn));
+  for (auto &&update_fn : _update_funcs)
+    ret->append(std::move(update_fn));
+  _update_funcs.clear();
```

Comment:
Why do you reset the `_update_funcs`? Do you consider the case of repeating backwarding several times?

Reply:
It's my mistake. It is not necessary.

Reply:
This method is called for every node to be generated a `TrainableFnSequence`. So `_update_funcs` must be empty before this method called for the next node.